core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/collection/WithinExpression.kt

```diff
@@ -17,29 +18,30 @@ class WithinExpression<T : ValidType>(
     override fun toDopeQuery(manager: DopeQueryManager) = toInfixDopeQuery(manager = manager)
 }
 
-fun <T : ValidType> TypeExpression<T>.withinArray(array: TypeExpression<ArrayType<T>>): WithinExpression<T> =
+fun <T : ValidType> TypeExpression<T>.withinArray(array: TypeExpression<ArrayType<T>>) =
     WithinExpression(this, array)
 
-fun Number.withinArray(array: TypeExpression<ArrayType<NumberType>>): WithinExpression<NumberType> =
-    toDopeType().withinArray(array)
+fun Number.withinArray(array: TypeExpression<ArrayType<NumberType>>) = toDopeType().withinArray(array)
 
-fun String.withinArray(array: TypeExpression<ArrayType<StringType>>): WithinExpression<StringType> =
-    toDopeType().withinArray(array)
+fun String.withinArray(array: TypeExpression<ArrayType<StringType>>) = toDopeType().withinArray(array)
 
-fun Boolean.withinArray(array: TypeExpression<ArrayType<BooleanType>>): WithinExpression<BooleanType> =
-    toDopeType().withinArray(array)
+fun Boolean.withinArray(array: TypeExpression<ArrayType<BooleanType>>) = toDopeType().withinArray(array)
 
-fun <T : ValidType> TypeExpression<T>.withinArray(array: Collection<TypeExpression<T>>): WithinExpression<T> =
-    this.withinArray(array.toDopeType())
+fun <T : ValidType> TypeExpression<T>.withinArray(array: Collection<TypeExpression<T>>) = withinArray(array.toDopeType())
 
-fun Number.withinArray(array: Collection<TypeExpression<NumberType>>): WithinExpression<NumberType> =
-    withinArray(array.toDopeType())
+fun Number.withinArray(array: Collection<TypeExpression<NumberType>>) = withinArray(array.toDopeType())
 
-fun String.withinArray(array: Collection<TypeExpression<StringType>>): WithinExpression<StringType> =
-    withinArray(array.toDopeType())
+fun String.withinArray(array: Collection<TypeExpression<StringType>>) = withinArray(array.toDopeType())
 
-fun Boolean.withinArray(array: Collection<TypeExpression<BooleanType>>): WithinExpression<BooleanType> =
-    withinArray(array.toDopeType())
+fun Boolean.withinArray(array: Collection<TypeExpression<BooleanType>>) = withinArray(array.toDopeType())
+
+fun <T : ValidType> TypeExpression<T>.withinArray(array: ISelectOffsetClause<T>) = withinArray(array.asExpression())
```

Comment:
now the name 'array' does not fit anymore. Would change it to _selectClause: ISelectOffsetClause_

Reply:
done